File: solutions/python/bob/1/bob.py
def response(hey_bob):
    # Remove whitespace from the ends
    phrase = hey_bob.strip()

    # 1. Silence
    if phrase == "":
        return "Fine. Be that way!"

    # Check if it's yelling (all letters are uppercase)
    is_yelling = phrase.isupper()
    # Check if it's a question (ends with '?')
    is_question = phrase.endswith("?")

    # 2. Yelled question
    if is_yelling and is_question:
        return "Calm down, I know what I'm doing!"

    # 3. Yelling
    if is_yelling:
        return "Whoa, chill out!"

    # 4. Question
    if is_question:
        return "Sure."

    # 5. Anything else
    return "Whatever."

File: solutions/java/bob/1/src/main/java/Bob.java
class Bob {
    String hey(String input) {
        String trimmed = input.trim();

        boolean isSilence = trimmed.isEmpty();
        boolean isQuestion = trimmed.endsWith("?");
        boolean isYelling = trimmed.equals(trimmed.toUpperCase()) && !trimmed.equals(trimmed.toLowerCase());

        if (isSilence) {
            return "Fine. Be that way!";
        }
        if (isYelling && isQuestion) {
            return "Calm down, I know what I'm doing!";
        }
        if (isYelling) {
            return "Whoa, chill out!";
        }
        if (isQuestion) {
            return "Sure.";
        }
        return "Whatever.";
    }
}